File: src/cviz/control/tcp/ClientAction.java
package cviz.control.tcp;

import java.io.Serializable;
import java.util.HashMap;

public class ClientAction implements Serializable {
    private static final long serialVersionUID = 8403870393123567760L;

    public enum ActionType {
        LOAD,
        CUE,
        KILL,
        QUERY
    }

    private String channel;
    private String name;
    private ActionType type;
    private String timelineFile;
    private String instanceName;
    private HashMap<String, String> parameters;

    public ClientAction(String channel, String name, ActionType type, String timelineFile, String instanceName, HashMap<String, String> parameters) {
        this.channel = channel;
        this.name = name;
        this.type = type;
        this.timelineFile = timelineFile;
        this.instanceName = instanceName;
        this.parameters = parameters;
    }

    public String getChannel() {
        if (channel == null || channel.length() == 0)
            return "default";

        return channel;
    }
    public String getName() {
        if (name == null || name.length() == 0)
            return "default";

        return name;
    }

    public ActionType getType() {
        return type;
    }

    public String getTimelineFile() {
        return timelineFile;
    }

    public String getInstanceId() {
        return instanceName;
    }

    public HashMap<String, String> getParameters() {
        return parameters;
    }

    public String toString() {
        return "Action: " + type + " " + timelineFile + " data: " + instanceName + " " + (parameters != null ? parameters.size() : "-");
    }
}

File: CViz/Control/Tcp/ClientAction.cs
using System.Collections.Generic;

namespace CViz.Control.Tcp
{
    class ClientAction
    {
        public enum ActionType
        {
            Unknown,
            Load,
            Cue,
            RunChild,
            Kill,
            Query,
        }

        public string TimelineSlot { get; set; }
        public ActionType Type { get; set; }
        public string TimelineFile { get; set; }
        public string InstanceName { get; set; }
        public Dictionary<string, string> Parameters { get; set; }

        public ClientAction()
        {
            TimelineSlot = "default";
        }

        public override string ToString()
        {
            string paramCount = Parameters?.Count.ToString() ?? "-";
            return $"Action: {TimelineSlot} {Type} {TimelineFile} data: {InstanceName} {paramCount}";
        }
    }
}
